jdaviz/app.py

```diff
@@ -614,12 +614,14 @@ def _on_layers_changed(self, msg):
             if children_layers is not None:
                 parent_icon = self.state.layer_icons[layer_name]
                 for i, child_layer in enumerate(children_layers, start=1):
-                    child_layer_icons[child_layer] = f'{parent_icon}{i}'
+                    if child_layer not in self.state.layer_icons:
+                        child_layer_icons[child_layer] = f'{parent_icon}{i}'
 
-        self.state.layer_icons = {
-            **self.state.layer_icons,
-            **child_layer_icons
-        }
+        if child_layer_icons:
+            self.state.layer_icons = {
+                **self.state.layer_icons,
+                **child_layer_icons
+            }
```

Comment:
FYI: @bmorris3 - does this seem reasonable?  I'm trying to avoid an update to the state again if no new child layers are added.